torchdrivesim/simulator.py

```diff
@@ -411,7 +418,7 @@ def compute_traffic_lights_violations(self) -> TensorPerAgentType:
         return violation
 
     @abc.abstractmethod
-    def _compute_collision_of_single_agent(self, box: Tensor, remove_self_overlap: bool = True) -> Tensor:
+    def _compute_collision_of_single_agent(self, box: Tensor, remove_self_overlap: Tensor = None, agent_types: list = []) -> Tensor:
```

Comment:
If the `Tensor` argument is optional, it should be `Optional[Tensor]`. The last argument should be `agent_types: Optional[List[str]] = None`. Mutable objects, such as lists, should not be used as defaults, since they don't get regenerated at each call. Also, I suspect the default means "all agent types", rather than "no agent types", so using an empty list for that is not consistent.

Reply:
Done.